crates/oxc_linter/src/config/oxlintrc.rs

```diff
@@ -141,6 +142,59 @@ impl Oxlintrc {
 
         Ok(config)
     }
+
+    /// Merges two [Oxlintrc] files together
+    /// [Self] takes priority over `other`
+    #[must_use]
+    pub fn merge(&self, other: Oxlintrc) -> Oxlintrc {
+        let mut categories = other.categories.clone();
+        categories.extend(self.categories.iter());
+
+        let rules = self
+            .rules
+            .rules
+            .iter()
+            .chain(&other.rules.rules)
+            .fold(FxHashMap::default(), |mut rules_set, rule| {
+                if rules_set.contains_key(&(&rule.plugin_name, &rule.rule_name)) {
+                    return rules_set;
+                }
+                rules_set.insert((&rule.plugin_name, &rule.rule_name), rule);
+                rules_set
+            })
+            .values()
+            .map(|rule| (**rule).clone())
+            .collect::<Vec<_>>();
+
+        let settings = self.settings.clone();
+        let env = self.env.clone();
+        let globals = self.globals.clone();
+
+        let mut overrides = self.overrides.clone();
+        overrides.extend(other.overrides);
+
+        Oxlintrc {
+            plugins: self.plugins.map_or_else(
+                || other.plugins,
+                |p| Some(other.plugins.map_or_else(|| p, |p2| p2.union(p))),
+            ),
+            categories,
+            rules: OxlintRules::new(rules),
+            settings,
+            env,
+            globals,
+            overrides,
+            path: self.path.clone(),
+            ignore_patterns: self
```

Comment:
I do not think this will work as expected.
When `folder/xyz/` has `**/ignore`, the `root` should not ignore the folder.

Problems will happen too when ignore pattern starts with `/`.

Reply:
yes you're right i need to normalise the paths and make them absolute relative to the config file i think

Reply:
@camc314 Will approve the changes when we do not append `ignore_patterns` :) 
This does not work in main and can be look later at. Ignoring is a bigger problem (see #11067)

The tests can be done later too.

Reply:
thanks 🙏 fixed in the latest revision

the existsing e2e tests _should_ cover this, but i agree we can always write more